Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1236,26 +1245,23 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
     }
 
     @available(iOS 17, *)
-    public func handleShutDown(completionHandler: ((Data?) -> Void)? = nil) {
-        Task { @MainActor in
-            let managers = try await NETunnelProviderManager.loadAllFromPreferences()
-
-            guard let manager = managers.first else {
-                completionHandler?(nil)
-                return
-            }
-
-            os_log("⚪️ Disabling Connect On Demand and shutting down the tunnel", log: .networkProtection)
+    @MainActor
+    public func handleShutDown() async throws {
+        os_log("🔴 Disabling Connect On Demand and shutting down the tunnel", log: .networkProtection)
+        let managers = try await NETunnelProviderManager.loadAllFromPreferences()
 
-            manager.isOnDemandEnabled = false
-            try await manager.saveToPreferences()
-            try await manager.loadFromPreferences()
+        guard let manager = managers.first else {
+            os_log("Could not find a viable manager, bailing out of shutdown", log: .networkProtection)
+            // Doesn't matter a lot what error we throw here, since we'll try cancelling the
+            // tunnel.
+            throw TunnelError.vpnAccessRevoked
```

Comment:
If we can't shut down the tunnel, we return an error so the parent method can try to cancel the tunnel.